components/PronunciationChecker/index.tsx

```
@@ -0,0 +1,207 @@
+    );
+  }
+
+  // Xử lý khi transcript thay đổi
+  useEffect(() => {
+    if (transcript && !listening && targetText) {
+      evaluatePronunciation(transcript, targetText);
+    }
+  }, [transcript, listening, targetText]);
+
+  // Cập nhật trạng thái nghe dựa trên SpeechRecognition
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)